crates/cdk/src/wallet/send.rs

```diff
@@ -45,135 +37,340 @@ impl Wallet {
-                        .await?
-                }
-            }
+        // Get active keyset ID
+        let active_keyset_id = self.get_active_mint_keyset().await?.id;
+        tracing::debug!("Active keyset ID: {:?}", active_keyset_id);
 
-            // Handle offline tolerance
-            (SendKind::OfflineTolerance(tolerance), Ok(selected_proofs), _) => {
-                let selected_proofs_amount = selected_proofs.total_amount()?;
+        // Get keyset fees
+        let keyset_fee_ppk = self.get_keyset_fees_by_id(active_keyset_id).await?;
+        tracing::debug!("Keyset fees: {:?}", keyset_fee_ppk);
 
-                let amount_to_send = match include_fees {
-                    true => amount + self.get_proofs_fee(&selected_proofs).await?,
-                    false => amount,
-                };
-                if selected_proofs_amount - amount_to_send <= *tolerance {
-                    selected_proofs
-                } else {
-                    tracing::info!("Selected proofs greater than tolerance. Must swap online");
-                    return Err(Error::InsufficientFunds);
-                }
+        // Calculate total send amount
+        let total_send_amount = send.amount + send.send_fee;
+        tracing::debug!("Total send amount: {}", total_send_amount);
+
+        // Swap proofs if necessary
+        if !send.proofs_to_swap.is_empty() {
+            let swap_amount = total_send_amount - proofs_to_send.total_amount()?;
+            tracing::debug!("Swapping proofs; swap_amount={:?}", swap_amount);
+            if let Some(proofs) = self
+                .swap(
+                    Some(swap_amount),
+                    SplitTarget::None,
+                    send.proofs_to_swap,
+                    send.options.conditions,
+                    false, // already included in swap_amount
+                )
+                .await?
+            {
+                proofs_to_send.extend(proofs);
             }
+        }
+        tracing::debug!(
+            "Proofs to send: {:?}",
+            proofs_to_send.iter().map(|p| p.amount).collect::<Vec<_>>()
+        );
 
-            // Handle online tolerance when selection fails and conditions are present
-            (SendKind::OnlineTolerance(_), Err(_), Some(_)) => {
-                tracing::info!("Could not select proofs with conditions while offline.");
-                tracing::info!("Attempting to select proofs without conditions and swapping");
+        // Check if sufficient proofs are available
+        if send.amount > proofs_to_send.total_amount()? {
+            return Err(Error::InsufficientFunds);
+        }
 
-                self.swap_from_unspent(amount, conditions, include_fees)
-                    .await?
-            }
+        // Update proofs state to pending spent
+        self.localstore
+            .update_proofs_state(proofs_to_send.ys()?, State::PendingSpent)
+            .await?;
```

Comment:
Should we check the proof state here to make sure they are Reserved or Unspent and have not already been used in another tx

Reply:
Maybe some new error ProofStateUnexpected and just add a log message.